src/lib.rs

```diff
@@ -216,7 +263,16 @@ pub fn build_cosmogony(
         .context("invalid osm file")?;
     info!("reading pbf done.");
 
-    let (mut zones, mut stats) = get_zones_and_stats(&parsed_pbf)?;
+    info!("Reading postal codes");
+    let file = File::open(&path).context("no pbf file")?;
+    let parsed_postal_code = OsmPbfReader::new(file)
+        .get_objs_and_deps(|o| is_postal_code(o))
+        .context("invalid osm file")?;
```

Comment:
@amatissart I kinda dislike that I have to parse the PBF file twice -- can we somehow do this in a single loop? My rust knowledge was not enough there ;-)

Reply:
I don't see why we couldn't extend the existing reading and simply write
```rust
let parsed_pbf = OsmPbfReader::new(file)
        .get_objs_and_deps(|o| is_admin(o) || is_place(o) || is_postal_code(o))
        .context("invalid osm file")?;
```

Then after a single call, `parsed_pbf` contains in memory all objects required by the rest of the process. And as long as you take care to filter the exact items that you need when iterating on its values, everything should be ok.

Reply:
I see! Cool! :)

Reply:
I think I solved this now :)